app/src/main/java/org/xbmc/kore/ui/volumecontrollers/VolumeControllerDialogFragment.java

```diff
@@ -0,0 +1,172 @@
+package org.xbmc.kore.ui.volumecontrollers;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.Nullable;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatDialogFragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.xbmc.kore.R;
+import org.xbmc.kore.host.HostConnectionObserver;
+import org.xbmc.kore.host.HostManager;
+import org.xbmc.kore.jsonrpc.ApiCallback;
+import org.xbmc.kore.jsonrpc.ApiMethod;
+import org.xbmc.kore.jsonrpc.method.Application;
+import org.xbmc.kore.ui.widgets.HighlightButton;
+import org.xbmc.kore.ui.widgets.VolumeLevelIndicator;
+import org.xbmc.kore.utils.LogUtils;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+
+public class VolumeControllerDialogFragment extends AppCompatDialogFragment
+        implements HostConnectionObserver.ApplicationEventsObserver,
```

Comment:
I would appreciate if we could minimize the use of the HostConnectionObserver. 
In this case i'm not sure it is strictly needed. It seems that you use it to get a callback about the volume changed, but that might be done differently. I'm not sure which way is better, so this can be in for now, i need to think a bit more on it.